extends CharacterBody3D
@onready var animated_sprite_3d: AnimatedSprite3D = $AnimatedSprite3D


const SPEED = 5.0
const JUMP_VELOCITY = 4.5


func _physics_process(delta: float) -> void:
	# Add the gravity.
	if not is_on_floor():
		velocity += get_gravity() * delta
	if is_on_floor() == false:
		animated_sprite_3d.animation = "falling"
	# Handle jump.
	if Input.is_action_just_pressed("Jump") and is_on_floor():
		velocity.y = JUMP_VELOCITY

	# Get the input direction and handle the movement/deceleration.
	# As good practice, you should replace UI actions with custom gameplay actions.
	var input_dir := Input.get_vector("Left", "Right", "Forward", "Backward")
	var direction := (transform.basis * Vector3(input_dir.x, 0, input_dir.y)).normalized()
	if direction:
		animated_sprite_3d.animation = "walking"
		velocity.x = direction.x * SPEED
		velocity.z = direction.z * SPEED
		if direction.x >= 0:
			animated_sprite_3d.flip_h = false
		if direction.x <= -0.5:
			animated_sprite_3d.flip_h = true
	else:
		animated_sprite_3d.animation = "idle"
		velocity.x = move_toward(velocity.x, 0, SPEED)
		velocity.z = move_toward(velocity.z, 0, SPEED)

	move_and_slide()
